lib/contextual_logger/logger_with_context.rb

```diff
@@ -16,7 +23,13 @@ def initialize(logger, context, level: nil)
       @logger = logger
       self.level = level
       @context = normalize_context(context)
-      @merged_context_cache = {}  # so we don't have to merge every time
+    end
+
```

Comment:
@jebentier This is still a bug. I realized it's one of the reasons I had headed down the path of tracking the `global_context` instance for `equals?`--that would be necessary in order to fix this bug. But it would be really complicated. And this bug isn't new. The code in this branch works much better in that it handles `with_context` on both the original logger as well as the `LoggerWithContext`; it just doesn't handle if you repeat the former after having logged something. That seems very unlikely.

Reply:
I agree that this seems unlikely and that the code required to fix this limitation is more complex than we want for this logger. Thank you for documenting it so we’re aware moving forward.